lib/IDE/CursorInfo.cpp

```diff
@@ -273,13 +276,33 @@ class CursorInfoTypeCheckSolutionCallback : public TypeCheckCompletionCallback {
   };
 
 private:
-  /// The expression for which we want to provide cursor info results.
-  Expr *ResolveExpr;
+  /// The source file and location to resolve.
+  /// Note that we cannot store the expression to resolve directly because an
+  /// \c UnresolvedDeclRefExpr might be replaced by an \c OverloadedDeclRefExpr
+  /// and thus the constraint system solution doesn't know about the
+  /// \c UnresolvedDeclRefExpr. Instead, we find the expression to resolve in
+  /// the source file again after expression pre-check has run.
+  SourceFile &SrcFile;
```

Comment:
Can this be the decl context instead? That should be faster.

Reply:
Good idea.